structs.go

```diff
@@ -1566,6 +1566,17 @@ type UserGuildSettingsEdit struct {
 	ChannelOverrides     map[string]*UserGuildSettingsChannelOverride `json:"channel_overrides"`
 }
 
+// A GuildMemberEditData stores a payload to use in Session.GuildMemberEdit
+// https://discord.com/developers/docs/resources/guild#modify-guild-member
+type GuildMemberEditData struct {
+
+	// Value to set user's nickname to
+	Nick string `json:"nick,omitempty"`
+
+	// Array of role ids the member is assigned
+	Roles []string `json:"roles,omitempty"`
```

Comment:
Since `roles` are optional, I think it would be best to make it `*[]string`.
As alternative we can put up a comment with notice about usage of `GuildMemberRoles` to reset member roles, but I think the first approach is better in this case.